### ISSUE TITLE
`jax.scipy.special.xlogy` and `xlog1py` return `0.0` instead of `nan` when `x=0` and `y=nan`

### ISSUE THREAD

**Issue by seanonymoussubmission** [jax-ml/jax#37751]
### Description

`jax.scipy.special.xlogy(x, y)` and `jax.scipy.special.xlog1py(x, y)` are supposed to compute `x * log(y)` and `x * log(1+y)` respectively, with the special case that the result is `0` when `x == 0` (to handle `0 * log(0)` gracefully).

However, when `x=0` and `y=nan`, JAX returns `0.0` instead of `nan`. This is incorrect — a `nan` input should propagate to the output because the mathematical expression `0 * log(nan)` is indeterminate. `scipy.special.xlogy` correctly returns `nan` in this case.

This breaks any downstream computation that depends on NaN propagation to detect invalid inputs — for example, log-likelihood computations in probabilistic models where `nan` in parameters should surface as `nan` in the loss.

## Minimal reproducer

```python
import jax.numpy as jnp
from jax.scipy import special as lsp_special
import scipy.special as osp_special
import numpy as np

x = jnp.float32(0.0)
y = jnp.float32(float("nan"))

print(osp_special.xlogy(0.0, float("nan")))   # nan  ← correct
print(lsp_special.xlogy(x, y))                 # 0.0  ← BUG: should be nan

print(osp_special.xlog1py(0.0, float("nan")))  # nan  ← correct
print(lsp_special.xlog1py(x, y))               # 0.0  ← BUG: should be nan
```

## Expected behavior

```
nan
nan
nan
nan
```

## Actual behavior

```
nan
0.0
nan
0.0
```

## Root cause

The JAX implementation short-circuits to `0` whenever `x == 0`, without first checking whether `y` is `nan`. The correct behavior is to return `nan` if either argument is `nan`, and only return `0` for `x == 0` when `y` is a valid (non-NaN) value. `scipy.special.xlogy` follows this contract.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:     0.10.1.dev20260513+64f6e229b
jaxlib:  0.10.0
numpy:   2.4.4
scipy:   1.17.1
python:  3.12.3
OS:      Linux-6.14.0-37-generic-x86_64-with-glibc2.39
backend: cpu
```
